Documentation/PackageRegistryUsage.md

```diff
@@ -0,0 +1,543 @@
+$ swift package-registry set https://packages.example.com 
+```
+
+The above sets registry to `https://packages.example.com` at project level. Pass 
+the `--global` option to set registry at user level:
+
+```bash
+$ swift package-registry set --global https://packages.example.com 
+```
+
+The resulting `registries.json` would look something like:
+
+```json
+{
+  "registries" : {
+    "[default]" : {
+      "url": "https://packages.example.com"
+    }   
+  },
+  "version" : 1
+}
+```
+
+The JSON key `[default]` means that the registry at `https://packages.example.com` is 
+"unscoped" and will be applied when there is no registry association found for 
+a given scope. 
+
+In this example, `https://packages.example.com` will be applied to all scopes.
+
+### Adding a registry package dependency
+
+A registry package dependency is declared in `Package.swift` using the
+package identifier. For example: 
+
+```swift
+dependencies: [
+    .package(id: "mona.LinkedList", .upToNextMajor(from: "1.0.0")),
+],
+```
+
+SwiftPM will query the registry mapped to a package's scope to 
+resolve and download the appropriate release version.  
+
+### Registry authentication
+
+If a registry requires authentication, it can be set up by using the 
+[`swift package-registry login` subcommand](https://github.com/apple/swift-evolution/blob/main/proposals/0378-package-registry-auth.md#new-login-subcommand) 
+introduced by SE-0378:
+
+```bash
+$ swift package-registry login
+OVERVIEW: Log in to a registry
+
+USAGE: swift package-registry login [<url>] [--username <username>] [--password <password>] [--token <token>] [--no-confirm]
+
+ARGUMENTS:
+  <url>                   The registry URL
+
+OPTIONS:
+  --username <username>   Username
+  --password <password>   Password
+  --token <token>         Access token
+  --no-confirm            Allow writing to netrc file without confirmation
+```
+
+Currently, basic and token authentication are supported.
+
+Provide the credentials either by setting the corresponding options
+(i.e., one of username/password or access token) or when prompted:
+
+```bash
+$ swift package-registry login https://packages.example.com
+```
+
+SwiftPM will save the credentials to the operating system's credential store
+(e.g., Keychain in macOS) or netrc file (which by default is located at `~/.netrc`) 
+and apply them automatically when making registry API requests.
+
+## Dependency Resolution Using Registry
+
+Resolving a registry dependency involves these steps:
+1. Fetch a package's available versions by calling the [list package releases](https://github.com/apple/swift-package-manager/blob/main/Documentation/Registry.md#41-list-package-releases) API.
+2. Compute the dependency graph by [fetching manifest(s) for a package release](https://github.com/apple/swift-package-manager/blob/main/Documentation/Registry.md#43-fetch-manifest-for-a-package-release).
+3. Pinpoint the package version to use.
+
+### Using registry for source control dependencies
```

Comment:
we should probably add some prose here on when this is useful, for example a mixed graph would fail to build given symbol clash across duplicate packages with different origins

Reply:
Reworked this section and added the info as requested